### ISSUE TITLE
include\node\common.gypi not found

### ISSUE THREAD

**Issue by dpilarsk** [electron/electron-rebuild#382]
Node version: 10.15.3
Electron version: 8.5.0
node-gyp version: 7.1.0
node-abi: 2.19.0
electron-rebuild: 2.0.0
node-keytar: 6.0.1
Platform: Windows x64

Hi 👋,

During the `node-keytar` rebuild step I'm getting an error:
```
$ electron-rebuild -f -w keytar
- Searching dependency tree
– Rebuild Failed
An unhandled error occurred inside electron-rebuild[39m
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.0
gyp info using node@10.15.3 | win32 | x64
gyp info find Python using Python version 3.7.3 found at "C:\Python37-32\python3.exe"
gyp info find VS using VS2017 (15.9.28307.1216) found at:
gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Python37-32\python3.exe
gyp info spawn args [ 'C:\\folder\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2',
gyp info spawn args   '-Dnode_gyp_dir=C:\\folder\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\.electron-gyp\\\\4.0.2\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\folder\\node_modules\\keytar',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp: C:\Windows\system32\config\systemprofile\.electron-gyp\4.0.2\include\node\common.gypi not found (cwd: C:\folder\node_modules\keytar) while reading includes of binding.gyp while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\folder\node_modules\node-gyp\lib\configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.19041
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\folder\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=8.0.0" "--arch=x64" "--dist-url=https://www.electronjs.org/headers" "--build-from-source"
gyp ERR! cwd C:\folder\node_modules\keytar
gyp ERR! node -v v10.15.3
gyp ERR! node-gyp -v v7.1.0
gyp ERR! not ok 

Failed with exit code: 1

Error: gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.0
gyp info using node@10.15.3 | win32 | x64
gyp info find Python using Python version 3.7.3 found at "C:\Python37-32\python3.exe"
gyp info find VS using VS2017 (15.9.28307.1216) found at:
gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Python37-32\python3.exe
gyp info spawn args [ 'C:\\folder\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2',
gyp info spawn args   '-Dnode_gyp_dir=C:\\folder\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\.electron-gyp\\\\4.0.2\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\folder\\node_modules\\keytar',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp: C:\Windows\system32\config\systemprofile\.electron-gyp\4.0.2\include\node\common.gypi not found (cwd: C:\folder\node_modules\keytar) while reading includes of binding.gyp while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\folder\node_modules\node-gyp\lib\configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.19041
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\folder\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=8.0.0" "--arch=x64" "--dist-url=https://www.electronjs.org/headers" "--build-from-source"
gyp ERR! cwd C:\folder\node_modules\keytar
gyp ERR! node -v v10.15.3
gyp ERR! node-gyp -v v7.1.0
gyp ERR! not ok 

Failed with exit code: 1
    at SafeSubscriber._error (C:\folder\node_modules\spawn-rx\lib\src\index.js:267:84)
    at SafeSubscriber.__tryOrUnsub (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.error (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:156:26)
    at Subscriber._error (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:92:26)
    at Subscriber.error (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:72:18)
    at MapSubscriber.Subscriber._error (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:92:26)
    at MapSubscriber.Subscriber.error (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:72:18)
    at SafeSubscriber._next (C:\folder\node_modules\spawn-rx\lib\src\index.js:242:65)
    at SafeSubscriber.__tryOrUnsub (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.next (C:\folder\node_modules\spawn-rx\node_modules\rxjs\internal\Subscriber.js:143:22)[39m
```

I can't reproduce the problem with Electron 7.3.2

Do you have any idea or knowledge about how to solve the problem ?
Thanks in advance for your help

**Comment by malept** [comment]
Please try again with 2.0.1, the error should be more obvious.

**Comment by dpilarsk** [comment]
This what I have after the upgrade

```
An unhandled error occurred inside electron-rebuild
Command failed with a non-zero return code (1):
C:\folder\node_modules\electron-rebuild\node_modules\.bin\node-gyp.cmd rebuild --target=8.0.0 --arch=x64 --dist-url=https://www.electronjs.org/headers --build-from-source

gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.0
gyp info using node@10.15.3 | win32 | x64
gyp info find Python using Python version 3.7.3 found at "C:\Python37-32\python3.exe"
gyp info find VS using VS2017 (15.9.28307.1216) found at:
gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Python37-32\python3.exe
gyp info spawn args [ 'C:\\folder\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2',
gyp info spawn args   '-Dnode_gyp_dir=C:\\folder\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\.electron-gyp\\\\4.0.2\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\folder\\node_modules\\keytar',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp: C:\Windows\system32\config\systemprofile\.electron-gyp\4.0.2\include\node\common.gypi not found (cwd: C:\folder\node_modules\keytar) while reading includes of binding.gyp while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\folder\node_modules\node-gyp\lib\configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.19041
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\folder\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=8.0.0" "--arch=x64" "--dist-url=https://www.electronjs.org/headers" "--build-from-source"
gyp ERR! cwd C:\folder\node_modules\keytar
gyp ERR! node -v v10.15.3
gyp ERR! node-gyp -v v7.1.0
gyp ERR! not ok

Error: Command failed with a non-zero return code (1):
C:\folder\node_modules\electron-rebuild\node_modules\.bin\node-gyp.cmd rebuild --target=8.0.0 --arch=x64 --dist-url=https://www.electronjs.org/headers --build-from-source

gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.0
gyp info using node@10.15.3 | win32 | x64
gyp info find Python using Python version 3.7.3 found at "C:\Python37-32\python3.exe"
gyp info find VS using VS2017 (15.9.28307.1216) found at:
gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Python37-32\python3.exe
gyp info spawn args [ 'C:\\folder\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\folder\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Windows\\system32\\config\\systemprofile\\.electron-gyp\\4.0.2',
gyp info spawn args   '-Dnode_gyp_dir=C:\\folder\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\.electron-gyp\\\\4.0.2\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\folder\\node_modules\\keytar',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\folder\\node_modules\\keytar\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp: C:\Windows\system32\config\systemprofile\.electron-gyp\4.0.2\include\node\common.gypi not found (cwd: C:\folder\node_modules\keytar) while reading includes of binding.gyp while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\folder\node_modules\node-gyp\lib\configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.19041
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\folder\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=8.0.0" "--arch=x64" "--dist-url=https://www.electronjs.org/headers" "--build-from-source"
gyp ERR! cwd C:\folder\node_modules\keytar
gyp ERR! node -v v10.15.3
gyp ERR! node-gyp -v v7.1.0
gyp ERR! not ok
    at ChildProcess.process.on (C:\folder\node_modules\@malept\cross-spawn-promise\dist\src\index.js:84:24)
    at ChildProcess.emit (events.js:189:13)
    at ChildProcess.cp.emit (C:\folder\node_modules\@malept\cross-spawn-promise\node_modules\cross-spawn\lib\enoent.js:34:29)
    at maybeClose (internal/child_process.js:970:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
```

**Comment by malept** [comment]
Try deleting `C:\Windows\system32\config\systemprofile\.electron-gyp` and try again.

**Comment by dpilarsk** [comment]
I still have the error after the delete

**Comment by malept** [comment]
It looks like you're running as Administrator or something? Usually the `.electron-gyp` folder is located in the user's home directory.

**Comment by dpilarsk** [comment]
No, I test on another computer where rebuild works and I don't have the `.electron-gyp` folder

**Comment by dpilarsk** [comment]
Yes, but with an other Electron version with the same setup works

**Comment by dpilarsk** [comment]
Issues fixed with 2.0.3 electron-rebuild version
